src/net/sourceforge/kolmafia/CoinmasterData.java

```diff
@@ -6,119 +6,662 @@
+  private BiConsumer<AdventureResult, Boolean> purchasedItem =
+      (x, y) -> purchasedItemInternal(x, y);
+
+  // Constructor for CoinmasterData with only mandatory fields.
+  // Optional fields can be added fluidly.
+  // Derived fields are built lazily
 
   public CoinmasterData(
-      final String master,
-      final String nickname,
-      final Class<? extends CoinMasterRequest> requestClass,
-      final String token,
-      final String tokenTest,
-      final boolean positiveTest,
-      final Pattern tokenPattern,
-      final AdventureResult item,
-      final String property,
-      final Map<Integer, Integer> itemRows,
-      final String buyURL,
-      final String buyAction,
-      final List<AdventureResult> buyItems,
-      final Map<Integer, Integer> buyPrices,
-      final String sellURL,
-      final String sellAction,
-      final List<AdventureResult> sellItems,
-      final Map<Integer, Integer> sellPrices,
-      final String itemField,
-      final Pattern itemPattern,
-      final String countField,
-      final Pattern countPattern,
-      final String storageAction,
-      final String tradeAllAction,
-      final boolean canPurchase) {
+      String master, final String nickname, final Class<? extends CoinMasterRequest> requestClass) {
     this.master = master;
     this.nickname = nickname;
     this.requestClass = requestClass;
+  }
+
+  // Fluid field construction
+
+  /**
+   * Defines the token used by the coinmaster,
+   *
+   * <p>A coinmaster deals in currencies other than Meat. If there is only one such currency, we
+   * call it the "token"
+   *
+   * @param token - Token used
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withToken(String token) {
     this.token = token;
+    return this;
+  }
+
+  /**
+   * Defines the plural name of the coinmaster's token.
+   *
+   * <p>The plural name is ordinarily lazily derived by adding "s" to the end of the token's name,
+   * or, if it is an item, using the item's plural name. If that does not suffice, specifying it
+   * here overrides lazy derivation.
+   *
+   * @param pluralToken - Plural name for the token.
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withPluralToken(String pluralToken) {
+    this.pluralToken = pluralToken;
+    return this;
+  }
+
+  /**
+   * Defines text that indicates you have no tokens available to spend.
+   *
+   * <p>When examining the responseText that shows the coinmaster's inventory, usually the available
+   * amount of each currency is displayed.
+   *
+   * <p>If there is only one currency - the token - and there is a simple string indicating that
+   * none are available, provide it here. If it is present, we need not do a pattern search to
+   * calculate the number of available tokens.
+   *
+   * <p>This is mutually exclusive with <code>positive</code>. Provide one or the other (or
```

Comment:
This seems wrong -- it's actually the case that this is a negative test by default, but if `positiveTest` is true it's a positive test.

Reply:
Right. fixed.